src/v/datalake/coordinator/state_machine.cc

```diff
@@ -41,6 +44,46 @@ void maybe_log_update_error(
 coordinator_stm::coordinator_stm(ss::logger& logger, raft::consensus* raft)
   : raft::persisted_stm<>("datalake_coordinator_stm.snapshot", logger, raft) {}
 
+ss::future<checked<model::term_id, coordinator_stm::errc>>
+coordinator_stm::sync(model::timeout_clock::duration timeout) {
+    auto sync_res = co_await ss::coroutine::as_future(
+      raft::persisted_stm<>::sync(timeout));
+    if (sync_res.failed()) {
+        auto eptr = sync_res.get_exception();
+        auto msg = fmt::format("Exception caught while syncing: {}", eptr);
+        if (ssx::is_shutdown_exception(eptr)) {
+            vlog(datalake_log.debug, "Ignoring shutdown error: {}", msg);
+            co_return errc::shutting_down;
+        }
+        vlog(datalake_log.warn, "{}", msg);
+        co_return errc::replication_error;
```

Comment:
this should probably be a timeout or some generic error? sync() doesn't replicate anything AFAIK.

Reply:
how about raft error? That's really what i mean by replication

Reply:
wfm.

Reply:
Done